pandas/core/indexes/base.py

```diff
@@ -2915,17 +2915,20 @@ def intersection(self, other):
             taken.name = None
         return taken
 
-    def difference(self, other):
+    def difference(self, other, sort=True):
         """
         Return a new Index with elements from the index that are not in
         `other`.
 
         This is the set difference of two Index objects.
-        It's sorted if sorting is possible.
 
         Parameters
         ----------
         other : Index or array-like
+        sort : bool, default True
+            Sort the resulting index if possible
+
+            .. versionadded:: 0.24.0
```

Comment:
can you make sure this is added to all subclasses as well (mutli, interval) I think have there own impl.

Reply:
can you do this (in this PR), can ideally update the tests for .difference for all types to parameterize it where appropriate